Cython/Compiler/Symtab.py

```diff
@@ -2622,6 +2622,7 @@ def declare_inherited_cpp_attributes(self, base_class):
                 base_entry.type, None, 'extern')
             entry.is_variable = 1
             entry.is_inherited = 1
+            entry.overloaded_alternatives = base_entry.overloaded_alternatives
```

Comment:
This should probably make a copy, in case the entry adds overloads – a case that seems worth testing, e.g. with two subclasses that add different overloads. That also suggests errors tests which you can add to `tests/errors/` with a `# mode: error`. See the other tests there on how to list the error output lines.

Reply:
Hi @scoder . I have updated my testcase in `tests/run/cpp_class_overloaded_traits.pyx`. As it stands, that test case works with the fix I made.

I don't completely understand what you mean by your suggestion. Can you give me just a small example based on my testcase that you want me to test if it works.

And the error tests you mentioned, also based on the sample test I have here on the PR, if you can just provide a snippet of something you reckon should fail, I will test it, and add a testcase.

Thanks in advance.

Reply:
Something like:

```
cdef:
    struct S1:
        pass

    struct S2:
        pass

	extern from "cpp_source.h":
		cppclass testclass:
			testclass()
            void a()
			void a(unsigned int i)

	cppclass testclass2(testclass):
		testclass2()

	cppclass testclass3(testclass):
		testclass3()
		void a(S1 v)

    cppclass testclass4(testclass):
        testclass4()
        void a(S2 v)

cdef S1 s1
cdef S2 s2
testclass4().a(s1)  # should fail
testclass4().a(s2)  # should pass
testclass3().a(s1)  # should pass
testclass3().a(s2)  # should fail
testclass2().a(s1)  # should fail
testclass2().a(s2)  # should fail
testclass1().a(s1)  # should fail
testclass1().a(s2)  # should fail
```

I think the worry is that because the entries share the same object for `overloaded_alternatives` that if you add an overload to one class then you add them to all classes.

Reply:
@da-woods Thanks a lot. I do not really know anything about c, c++ or Cython, so thanks a lot. I think I got everything you and @scoder wanted. Can you have a look and tell me if its alright?

Can we also run the workflows again?